### PR TITLE
Fix transitive deps

### PR DESCRIPTION
**Pull request by liam923**
Add support for `SH` and `BH` directives in `.merlin`, which allow specifying transient dependencies.

### CODE REVIEW

File: src/frontend/query_commands.ml [hunk 1]
```diff
@@ -827,7 +829,7 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
 
   | Path_list `Build ->
     let config = Mpipeline.final_config pipeline in
-    Mconfig.(config.merlin.build_path.visible @ config.merlin.build_path.hidden)
+    Mconfig.(config.merlin.build_path)
```
Comment (ccasin) [inline src/frontend/query_commands.ml:832]: I don't really know what this is used for, but why shouldn't it include the `-H` paths?

**Review by ccasin — APPROVED** [review]
Looks good.  I'm not an expert on every part of merlin touched here, and there are a couple bits I'm low confidence on, but the tests convince me this largely fixes the immediate problem in practice.  We should move forward and try to roll a fix quickly - I've put some comments and questions below, but it's fine to address them in a future PR.

File: tests/test-dirs/hidden-deps.t/run.t [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+Merlin can locate a value in an interface from a hidden dependency
+
+  $ $MERLIN single locate -position 1:11 -look-for interface -filename c/correct.ml < c/correct.ml | jq ".value"
+  {
+    "file": "$TESTCASE_ROOT/b/b.mli",
+    "pos": {
+      "line": 1,
+      "col": 0
+    }
+  }
```
[the remark this answers is not in the document]
Comment (liam923) [reply tests/test-dirs/hidden-deps.t/run.t:60]: I agree this is sad, but it seems to be consistent with how Merlin behaves for direct dependencies.

File: src/frontend/query_commands.ml [hunk 3]
```diff
@@ -181,8 +181,11 @@ let dump pipeline = function
 
   | [`String "paths"] ->
     let paths = Mconfig.build_path (Mpipeline.final_config pipeline) in
-    `Assoc [ "visible", `List (List.map paths.visible ~f:(fun s -> `String s));
-             "hidden", `List (List.map paths.hidden ~f:(fun s -> `String s)) ]
+    `List (List.map paths ~f:(fun s -> `String s))
+
+  | [`String "hidden-paths"] ->
```
[the remark this answers is not in the document]
Comment (liam923) [reply src/frontend/query_commands.ml:186]: This is just for debugging information so I think it's okay to skip tests for this. There's no tests covering this behavior for visible dependencies.

File: src/frontend/query_commands.ml (hunk 1 above)
Comment (liam923) [reply src/frontend/query_commands.ml:832]: I don't know what this is used for either, I think I just overlooked this. I think you're probably right that it should include hidden deps.

**Pull request merged**